packages/eslint-config/src/utils.ts

```diff
@@ -0,0 +1,57 @@
+import globals from 'globals';
+import { config as tseslintConfig } from 'typescript-eslint';
+
+import { plugins } from './plugins/index.js';
+
+/**
+ * Generate a plugins object from a list of ESLint plugin names
+ * (Only plugins that are known to `@rightcapital/eslint-config`).
+ *
+ * @see {@link plugins} for the list of plugins.
+ */
+export function pickPlugins(pluginNames?: Array<keyof typeof plugins>) {
+  if (!pluginNames) {
+    return plugins;
+  }
+
+  return Object.fromEntries(
+    pluginNames.map((pluginName) => [pluginName, plugins[pluginName]]),
+  );
+}
+
+const defineConfig: typeof tseslintConfig = (...configs) =>
+  tseslintConfig(...configs).map((config) => {
+    const knownPluginNames = Object.keys(plugins).filter((pluginName) =>
+      Object.keys(config.rules ?? {}).some((rule) =>
+        rule.startsWith(`${pluginName}/`),
+      ),
+    ) as Array<keyof typeof plugins>;
+    const resolvedPlugins = {
+      ...pickPlugins(knownPluginNames),
+      ...config.plugins,
+    };
+    return {
+      ...(Object.keys(resolvedPlugins).length > 0
```

Comment:
Is it necessary?

Reply:
Seems unnecessary since returning an empty plugins object would be safe even when there are no plugins.

Reply:
Yes, this is critical. We do not want every config object to include a `plugins` key, as doing so would break the global ignore config. For example, `defineConfig({ ignores: ['dist'] })` would fail if we omitted this logic.

ESLint requires that the global ignore config contain only an `ignores` key.( see https://eslint.org/docs/latest/use/configure/ignore#ignoring-files:~:text=key%20is%20used-,without%20any%20other%20keys%20in%20the%20configuration%20object,-%2C%20then%20the%20patterns)